persistence.js

```diff
@@ -346,6 +361,9 @@ function updateWithClientData (that, client, packet, cb) {
   })
 
   var ttl = that.packetTTL(packet)
+
+  if (!packet.messageId) delete packet.messageId
```

Comment:
never use `delete`. set it to `-1` or `undefined`.

Reply:
based on mqtt spec, we should undefine messageId if it is QoS 0.

Reply:
The problem is that abstract tests mostly fail because of messageId that both in mongo and redis becomes null after save into database (or encoded in case of redis). I could make it undefined after the decode

Reply:
`null` is also ok.

Reply:
delete slows everthing down.

Reply:
@mcollina Setting it to `null` breaks abstract tests on persistence

Reply:
delete removes a lot of V8 optimizations from our code. We should avoid this.

Reply:
I will fix this on aedes persistence so

Reply:
I think the best thing would be accept both null and undefined as valid. Would this be fine?